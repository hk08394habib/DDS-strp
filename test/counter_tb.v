`timescale 1us/1us

module test_counter();
   reg clk = 1'b0;
   reg reset = 1'b0;
   wire [3:0] phase_acc = 4'b0;

   phase_acc phase_acc_DUT(.clk(clk),
                           .mult(2),
                           .reset(reset),
                           .phase_acc(phase_acc));

   initial begin
      clk = 1'b0;
      forever #1 clk = ~clk;
   end

   initial begin
      $dumpfile("counter.vcd");
      $dumpvars();
      #30
        $finish();
   end

   always @(*) begin
      $monitor("[$monitor] time=%0t, phase_acc=%0d \n", $time, phase_acc_DUT.phase_acc);
   end

endmodule; // test_counter
