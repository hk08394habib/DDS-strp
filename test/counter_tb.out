#! /usr/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 6;
:vpi_module "/usr/lib64/ivl/system.vpi";
:vpi_module "/usr/lib64/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib64/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib64/ivl/v2005_math.vpi";
:vpi_module "/usr/lib64/ivl/va_math.vpi";
S_0x5579116e4790 .scope module, "sin_lookup" "sin_lookup" 2 3;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 8 "address";
    .port_info 2 /OUTPUT 8 "value";
o0x7f5be9ecf018 .functor BUFZ 8, C4<zzzzzzzz>; HiZ drive
v0x5579116e57b0_0 .net "address", 7 0, o0x7f5be9ecf018;  0 drivers
o0x7f5be9ecf048 .functor BUFZ 1, C4<z>; HiZ drive
v0x5579116f9a80_0 .net "clk", 0 0, o0x7f5be9ecf048;  0 drivers
o0x7f5be9ecf078 .functor BUFZ 8, C4<zzzzzzzz>; HiZ drive
v0x5579116f9b40_0 .net "value", 7 0, o0x7f5be9ecf078;  0 drivers
S_0x5579116e4970 .scope module, "test_counter" "test_counter" 3 4;
 .timescale -6 -6;
L_0x7f5be9e86060 .functor BUFT 1, C4<00000000000001111010000100100000>, C4<0>, C4<0>, C4<0>;
v0x5579116fa3d0_0 .net/2s *"_ivl_2", 31 0, L_0x7f5be9e86060;  1 drivers
v0x5579116fa4d0_0 .var "clk", 0 0;
L_0x7f5be9e86018 .functor BUFT 1, C4<0000>, C4<0>, C4<0>, C4<0>;
RS_0x7f5be9ecf2e8 .resolv tri, L_0x7f5be9e86018, L_0x55791170a8f0;
v0x5579116fa590_0 .net8 "phase_acc", 3 0, RS_0x7f5be9ecf2e8;  2 drivers
v0x5579116fa660_0 .var "reset", 0 0;
E_0x5579116aaf40 .event anyedge, v0x5579116fa1d0_0;
L_0x55791170a7d0 .part L_0x7f5be9e86060, 0, 20;
L_0x55791170a8f0 .part v0x5579116fa1d0_0, 0, 4;
S_0x5579116f9ca0 .scope module, "phase_acc_DUT" "phase_acc" 3 9, 4 16 0, S_0x5579116e4970;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "clk";
    .port_info 1 /INPUT 20 "f_out";
    .port_info 2 /INPUT 1 "reset";
    .port_info 3 /OUTPUT 8 "phase_acc";
P_0x5579116f9ea0 .param/l "f_clk" 0 4 24, +C4<00000000000011110100001001000000>;
v0x5579116fa010_0 .net "clk", 0 0, v0x5579116fa4d0_0;  1 drivers
v0x5579116fa0f0_0 .net "f_out", 19 0, L_0x55791170a7d0;  1 drivers
v0x5579116fa1d0_0 .var "phase_acc", 7 0;
v0x5579116fa290_0 .net "reset", 0 0, v0x5579116fa660_0;  1 drivers
E_0x5579116ce190 .event negedge, v0x5579116fa290_0;
E_0x5579116d3cc0 .event posedge, v0x5579116fa010_0;
    .scope S_0x5579116f9ca0;
T_0 ;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v0x5579116fa1d0_0, 0, 8;
    %end;
    .thread T_0;
    .scope S_0x5579116f9ca0;
T_1 ;
    %wait E_0x5579116d3cc0;
    %load/vec4 v0x5579116fa1d0_0;
    %pad/u 32;
    %pushi/vec4 1000000, 0, 32;
    %load/vec4 v0x5579116fa0f0_0;
    %pad/u 32;
    %div;
    %add;
    %pad/u 8;
    %assign/vec4 v0x5579116fa1d0_0, 0;
    %jmp T_1;
    .thread T_1;
    .scope S_0x5579116f9ca0;
T_2 ;
    %wait E_0x5579116ce190;
    %pushi/vec4 0, 0, 8;
    %assign/vec4 v0x5579116fa1d0_0, 0;
    %jmp T_2;
    .thread T_2;
    .scope S_0x5579116e4970;
T_3 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5579116fa4d0_0, 0, 1;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5579116fa660_0, 0, 1;
    %end;
    .thread T_3;
    .scope S_0x5579116e4970;
T_4 ;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0x5579116fa4d0_0, 0, 1;
T_4.0 ;
    %delay 1, 0;
    %load/vec4 v0x5579116fa4d0_0;
    %inv;
    %store/vec4 v0x5579116fa4d0_0, 0, 1;
    %jmp T_4.0;
    %end;
    .thread T_4;
    .scope S_0x5579116e4970;
T_5 ;
    %vpi_call 3 20 "$dumpfile", "counter.vcd" {0 0 0};
    %vpi_call 3 21 "$dumpvars" {0 0 0};
    %delay 30, 0;
    %vpi_call 3 23 "$finish" {0 0 0};
    %end;
    .thread T_5;
    .scope S_0x5579116e4970;
T_6 ;
    %wait E_0x5579116aaf40;
    %vpi_call 3 27 "$monitor", "[$monitor] time=%0t, phase_acc=%0d \012", $time, v0x5579116fa1d0_0 {0 0 0};
    %jmp T_6;
    .thread T_6, $push;
# The file index is used to find the file name in the following table.
:file_names 5;
    "N/A";
    "<interactive>";
    "src/sin_lookup.v";
    "./test/counter_tb.v";
    "src/counter.v";
